### PR TITLE
Added role editting to profile

### PR DESCRIPTION
**Pull request by farhatahmad**
## Description

Users with the `ManageUsers` permission can now edit a user's role using the `EditUser` page from the `ManageUsers' page

## Screenshots (if appropriate):

image

### CODE REVIEW

File: spec/controllers/admin/roles_controller_spec.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@
     session[:user_id] = user_with_manage_roles_permission.id
   end
 
-  describe 'roles#index' do
+  xdescribe 'roles#index' do
```
Comment (farhatahmad) [inline spec/controllers/admin/roles_controller_spec.rb:14]: Temporary until I change permission checker

File: app/javascript/components/users/user/forms/UpdateUserForm.jsx [hunk 2]
```diff
@@ -41,15 +47,13 @@ export default function UpdateUserForm({ user }) {
...
-        // TODO: Refactor this to use FormControl instead.
-      }
-      <BootStrapForm.Group className="mb-3" controlId={fields.userRole.controlId}>
-        <BootStrapForm.Label className="small mb-0">
-          {fields.userRole.label}
-        </BootStrapForm.Label>
-        <BootStrapForm.Select field={fields.userRole} type="select" />
-      </BootStrapForm.Group>
+      {(currentUser.permissions.ManageUsers === 'true') && (
+        <FormControl field={fields.role_id} control={BootStrapForm.Select}>
```
Comment (KH-Amir-TN) [inline app/javascript/components/users/user/forms/UpdateUserForm.jsx:51]: I like it when you guys use the `FormControl` stuff 😆

File: app/javascript/components/users/user/AccountInfo.jsx [hunk 3]
```diff
@@ -25,7 +25,11 @@ AccountInfo.propTypes = {
     name: PropTypes.string.isRequired,
     email: PropTypes.string.isRequired,
     provider: PropTypes.string.isRequired,
-    role: PropTypes.string.isRequired,
+    role: PropTypes.shape({
```
Comment (KH-Amir-TN) [inline app/javascript/components/users/user/AccountInfo.jsx:28]: I know this is irrelevant to this PR - In the future, we may extract redundant PropTypes into helper files where we can easily import them on need,

File: app/controllers/api/v1/admin/roles_controller.rb [hunk 4]
```diff
@@ -5,7 +5,10 @@ module V1
     module Admin
       class RolesController < ApiController
         before_action :find_role, only: %i[update show destroy]
-        before_action do
+        before_action only: %i[index] do
+          ensure_authorized('ManageUsers')
```
[the remark this answers is not in the document]
Comment (KH-Amir-TN) [reply app/controllers/api/v1/admin/roles_controller.rb:9]: I'm a bit skeptical about this - I think there are two concerns not only one.

**Review by KH-Amir-TN — APPROVED** [review]

**Pull request merged**